### PR TITLE
feat(gen-dto): add option to remove circular references from schemas

### PR DESCRIPTION
**Pull request by Mierenga**
This adds a `--remove-circular` option to the `dto` CLI which will find and replace circular references in schemas with a generic `{ type: 'object' }` json-schema.

In the future we could support recursive/circular schemas using https://ajv.js.org/guide/combining-schemas.html#combining-schemas, but for now we will allow removing them since they aren't critical for current use-cases.

### CODE REVIEW

File: src/cli/dto/gen-dtos.ts [hunk 1]
```diff
@@ -27,11 +33,88 @@ export class GenDTOs {
     return fileContent
   }
 
+  private static removeCircularReferences(obj: any): CircularRefInfo[] {
```
Comment (zekth) [inline src/cli/dto/gen-dtos.ts]: can we add tests for this?

Comment (Mierenga) [reply src/cli/dto/gen-dtos.ts]: https://github.com/Kong/brij/pull/2/commits/d42c737ce193d52393c09e1b9068fc261773b546

**Review by zekth — COMMENTED** [review]
Why is there `dist` `bin` commited? This should not be part of the codebase.

**Review by zekth — APPROVED** [review]
lgtm

**Review by zekth — APPROVED** [review]
lgtm

**Pull request merged**